### PR TITLE
adding and configuring  user location o map

### LINKED ISSUE #2: Realocate mocked pin on map

### PR DESCRIPTION
**Pull request by rainarodrigues**
#2 The User Location on a Map feature was created, with user privacy permissions options within the app.

### CODE REVIEW

File: SocialMap/Source/View/Core/UserFeed/UserFeedView.swift [hunk 1]
```diff
@@ -2,13 +2,13 @@ import SwiftUI
 import MapKit
 
 struct UserFeedView: View {
-    @ObservedObject private var controller = UserFeedController()
+    @StateObject private var controller = UserFeedController()
     
     var body: some View {
-        MapView ( coordinate: controller.userLocation ?? CLLocationCoordinate2D(latitude: 3.01, longitude: 4.0))
+        MapView ( coordinate: controller.userLocation?.center ?? CLLocationCoordinate2D(latitude: 3.01, longitude: 4.0))
```
Comment (ThiagoHBA) [inline SocialMap/Source/View/Core/UserFeed/UserFeedView.swift]: Usar o .init() igual o código que o @CapistranoIFCE tinha feito aqui vai melhorar na leitura

**Review by ThiagoHBA — CHANGES_REQUESTED** [review]

File: SocialMap/Source/View/Core/UserFeed/ MockedPinView.swift [hunk 2]
```diff
@@ -0,0 +1,8 @@
+//
```
Comment (daneccc) [inline SocialMap/Source/View/Core/UserFeed/ MockedPinView.swift:1]: remove header comments

**Review by daneccc — CHANGES_REQUESTED** [review]

File: SocialMap/Source/View/Core/UserFeed/UserFeedView.swift (hunk 1 above)
Comment (rainarodrigues) [reply SocialMap/Source/View/Core/UserFeed/UserFeedView.swift]: Adicionado

File: SocialMap/Source/View/Core/UserFeed/ MockedPinView.swift (hunk 2 above)
Comment (rainarodrigues) [reply SocialMap/Source/View/Core/UserFeed/ MockedPinView.swift:1]: Removido

**Pull request merged**